tangos/tools/property_importer.py

```diff
@@ -86,7 +89,12 @@ def _import_properties_for_timestep(self, ts, property_names, object_typetag):
         property_db_names = [core.dictionary.get_or_create_dictionary_item(self._session, name) for name in
                              property_names]
         rows_to_store = []
-        for values in self.handler.iterate_object_properties_for_timestep(ts.extension, object_typetag, property_names):
+        kwa = {}
+        # For AdaptaHOP handler, we do not need to load the entire snaphshot to enumerate
+        # properties in the halo catalogue
+        if isinstance(self.handler, RamsesAdaptaHOPInputHandler):
+            kwa["index_parent"] = False
```

Comment:
It seems weird to have logic here that changes what you pass to the `handler` based on what class it is. Can't you just locate the logic fully within `RamsesAdaptaHOPInputHandler` and so remove the need for any specific references to handler classes here?